scripts/setup-jest.ts

```diff
@@ -91,8 +133,41 @@ const getMockVSCode = () => {
       workspaceFolders: [],
       fs: {
         writeFile: jest.fn()
-      }
-    }
+      },
+      registerTextDocumentContentProvider: jest.fn()
+    },
+    CompletionItem: class {
+      public constructor(label: string) {}
+    },
+    CodeLens: class {
+      public constructor(range: Range) {}
+    },
+    DocumentLink: class {
+      public constructor(range: Range, target?: Uri) {}
+    },
+    CodeAction: class {
+      public constructor(title: string, data?: any) {}
+    },
+    Diagnostic: class {
+      public constructor(range: Range, message: string, severity?: any) {}
+    },
+    CallHierarchyItem: class {
+      public constructor(kind: any, name: string, detail: string, uri: Uri, range: Range, selectionRange: Range) {}
+    },
+    TypeHierarchyItem: class {
+      public constructor(kind: any, name: string, detail: string, uri: Uri, range: Range, selectionRange: Range) {}
+    },
+    SymbolInformation: class {
+      public constructor(name: string, kind: any, range: Range, uri?: Uri, containerName?: string) {}
+    },
+    InlayHint: class {
+      public constructor(position: any, label: any, kind?: any) {}
+    },
+    CancellationError: class {
+      public constructor() {}
+    },
```

Comment:
Is there a reason all these don't just point to a jest.fn()?  One downside of having a bunch of classes in here is if/when the VSCode API changes we'll have to stay in sync.